src/models/unit.ts

```diff
@@ -34,7 +42,24 @@ class Unit extends Model<InferAttributes<Unit>, InferCreationAttributes<Unit>> {
     return await Models.PlayerUnit.findAll({ where: { unitId: this.id } });
   }
 
-  canMakeRangedAttack(): Boolean {
+  toSnarkyUnit(): SnarkyUnit {
+    return new SnarkyUnit({
+      stats: new UnitStats({
+        health: UInt32.from(this.maxHealth),
+        movement: UInt32.from(this.movementSpeed),
+        rangedAttackRange: UInt32.from(this.rangedRange || 0),
+        rangedHitRoll: UInt32.from(this.rangedHitRoll || 0),
+        rangedWoundRoll: UInt32.from(this.rangedWoundRoll || 0),
+        saveRoll: UInt32.from(this.armorSaveRoll || 0),
+        rangedDamage: UInt32.from(this.rangedDamage || 0),
+        meleeHitRoll: UInt32.from(this.meleeHitRoll || 0),
+        meleeWoundRoll: UInt32.from(this.meleeWoundRoll || 0),
+        meleeDamage: UInt32.from(this.meleeDamage || 0),
```

Comment:
Should this also include `rangedNumAttacks`, `rangedArmorPiercing`, `rangedAmmo`, `meleeNumAttacks`, and `meleeArmorPiercing`? Or do these not exist on the Snarky type?

Reply:
They don't exist on the snarky type.  That won't stop the server from using logic based on them, and we can add them to the proof later.  But technically the circuit will not validate these values in the logic.

I still have work to do to get the turn and game transitions into the proof system itself.  But I wanted to start integrating with the server too so we don't build everything in a silo.